fvm/env.go

```diff
@@ -495,6 +499,12 @@ func (e *hostEnv) UpdateAccountContractCode(address runtime.Address, name string
 		return errors.New("updating account contract code is not supported")
 	}
 
+	// record updated contract
+	e.changedPrograms = append(e.changedPrograms, ChangedProgram{
+		Address: address,
+		Name:    name,
+	})
+
```

Comment:
you don't need this logic here, I added a method to env to notify of updates

Reply:
I think it would be good to collect names of changed contracts. Also, nothing in the master yet, lets see what we get with merges